lib/sensors/tof

```diff
@@ -0,0 +1,40 @@
+#include <iostream>
+#include<.clang-format>
+#include<i2c_sensors.hpp>
+#include "tof.hpp"
+
+namespace hyped::sensor{
+std::optional<TimeOfFlight> TimeOfFlight::create(core::ILogger &logger,
+                                                       std::shared_ptr<io::II2c> i2c,
+                                                       const std::uint8_t device_address)
+{ {
+  return TimeOfFlight(logger, i2c, device_address);
+}
+
+TimeOfFlight::TimeOfFlight    (core::ILogger &logger,
```

Comment:
some of this could be formatted a bit better, make sure you are using `clang-format` to format all your `.cpp` and `.hpp` files